scripts/queue_manager.R

```diff
@@ -66,24 +67,40 @@ list_doc <- function(type, conn) {
 }
 
 
+    
+lockfile <- file.path(get_database_dirname(), "lock")
+ndr_counter <- 0 # no data request counter, if 12 print msg in log
 
 while(TRUE) {
 
+    lfres <- lock(lockfile, timeout = 1000)
+    if(is.null(lfres)) {
+        log_debug("{logprefix} lock detected, sleeping ....")
+        Sys.sleep(floor(runif(1, min = 60, max = 120)))
+        next
+    } 
+    
+
     joblist <- list_doc(type = "data_req", conn = pool)
     if(length(joblist) == 0) {
-        log_trace("queue_man: no data requests")
+        ndr_counter <- ndr_counter + 1
```

Comment:
Why is this counter needed?

Reply:
To reduce the number of log messages (and the size of the log file). Only every 12th step  (i.e. once a minute) a message is printed